### PR TITLE
net: wifi: Display MFP info in scan results

### PR DESCRIPTION
**Pull request by rado17**
Display MFP info in scan results.

### CODE REVIEW

**Comment by rado17** [comment]
@sachinthegreen @ajayparida @krish2718 . Please review

**Review by krish2718 — DISMISSED** [review]

**Review by sachinthegreen — DISMISSED** [review]

**Comment by rlubos** [comment]
@rado17 There are compliance error:
>  LONG_LINE_STRING: line length of 106 exceeds 100 columns
File:subsys/net/l2/wifi/wifi_shell.c
Line:94
 LONG_LINE: line length of 102 exceeds 100 columns
File:subsys/net/l2/wifi/wifi_shell.c
Line:101

Please fix.

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -87,18 +87,21 @@ static void handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
...
 	      wifi_band_txt(entry->band),
 	      entry->rssi,
 	      wifi_security_txt(entry->security),
 	      ((entry->mac_length) ?
-		      net_sprint_ll_addr_buf(entry->mac, WIFI_MAC_ADDR_LEN, mac_string_buf,
-					     sizeof(mac_string_buf)) : ""));
+		      net_sprint_ll_addr_buf(entry->mac, WIFI_MAC_ADDR_LEN,
+								 mac_string_buf,
+								 sizeof(mac_string_buf)) : ""),
+								 wifi_mfp_txt(entry->mfp));
```
[the remark this answers is not in the document]
Comment (rado17) [reply subsys/net/l2/wifi/wifi_shell.c]: Done

File: subsys/net/l2/wifi/wifi_shell.c [hunk 2]
```diff
@@ -87,18 +87,21 @@ static void handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
...
 	      wifi_band_txt(entry->band),
 	      entry->rssi,
 	      wifi_security_txt(entry->security),
 	      ((entry->mac_length) ?
-		      net_sprint_ll_addr_buf(entry->mac, WIFI_MAC_ADDR_LEN, mac_string_buf,
-					     sizeof(mac_string_buf)) : ""));
+		      net_sprint_ll_addr_buf(entry->mac, WIFI_MAC_ADDR_LEN,
+					    mac_string_buf,
+					    sizeof(mac_string_buf)) : ""),
+					    wifi_mfp_txt(entry->mfp));
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:104]:
```suggestion
	      ((entry->mac_length) ?
		      net_sprint_ll_addr_buf(entry->mac, WIFI_MAC_ADDR_LEN, mac_string_buf,
					     sizeof(mac_string_buf)) : ""));
		      net_sprint_ll_addr_buf(entry->mac, WIFI_MAC_ADDR_LEN,
					     mac_string_buf,
					     sizeof(mac_string_buf)) : ""),
	      wifi_mfp_txt(entry->mfp));
```

Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:104]: This is `print` parameter, should be aligned with others.

File: subsys/net/l2/wifi/wifi_shell.c [hunk 3]
```diff
@@ -87,18 +87,21 @@ static void handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
...
 		print(context.sh, SHELL_NORMAL,
-		      "\n%-4s | %-32s %-5s | %-13s | %-4s | %-15s | %s\n",
-		      "Num", "SSID", "(len)", "Chan (Band)", "RSSI", "Security", "BSSID");
+		      "\n%-4s | %-32s %-5s | %-13s | %-4s | %-15s | %-17s | %-8s\n",
+		      "Num", "SSID", "(len)", "Chan (Band)", "RSSI", "Security", "BSSID", "MFP");
 	}
 
-	print(context.sh, SHELL_NORMAL, "%-4d | %-32s %-5u | %-4u (%-6s) | %-4d | %-15s | %s\n",
+	print(context.sh, SHELL_NORMAL,
+		 "%-4d | %-32s %-5u | %-4u (%-6s) | %-4d | %-15s | %-17s | %-8s\n",
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:95]:
```suggestion
	print(context.sh, SHELL_NORMAL,
	      "%-4d | %-32s %-5u | %-4u (%-6s) | %-4d | %-15s | %-17s | %-8s\n",
```

**Review by krish2718 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**